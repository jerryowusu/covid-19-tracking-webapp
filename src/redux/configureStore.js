import { combinedReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import reducer from './covidData';

const rootReducer = combinedReducers({
  reducer,
});

const store = createStore(
  rootReducer, applyMiddleware(thunk, logger),
);

export default store;
