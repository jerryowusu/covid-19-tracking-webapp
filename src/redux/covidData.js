import fetchData from 'services/fetchData';

const GET_DATA = 'GET_DATA';

const getCovidData = () => async (dispatch) => {
  try {
    const response = await fetchData();
    const covid = [];
    response.data.forEach((data) => {
      covid.push({
        data,
      });
    });
    dispatch({
      type: GET_DATA,
      covid,
    });
    return Promise.resolve(covid);
  } catch (err) {
    return Promise.reject(err);
  }
};


export { getCovidData };
export default reducer;
