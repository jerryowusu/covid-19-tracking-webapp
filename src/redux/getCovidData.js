const getDataFromAPI = async () => {
  const info = [];
  const response = await fetch('https://disease.sh/v3/covid-19/countries');
  const dataFromAPI = await response.json();

  dataFromAPI.map(({ countryInfo: { _id: id, flag }, ...data }) => {
    const covidData = {
      continent: data.continent,
      country: data.country,
      country_id: id,
      country_flag: flag,
      total_cases: data.cases,
      total_deaths: data.deaths,
      total_recovered: data.recovered,
      total_active: data.active,
      total_tests: data.tests,
      population: data.population,
      todays_cases: data.todayCases,
      todays_deaths: data.todayDeaths,
      todays_recovered: data.todayRecovered,
    };

    return info.push(covidData);
  });

  return info;
};

export default getDataFromAPI;
