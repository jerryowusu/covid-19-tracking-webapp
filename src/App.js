import { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { Route, Routes } from 'react-router-dom';
import { GlobalData } from './redux/getCovidData';
import Global from './components/Global';
import Navbar from './components/NavBar';
import Country from './components/Country';

const App = () => (
    <>
      <Navbar />
      <Routes>
        <Route exact path="/" element={<Global />} />
        <Route path="/country/:name" element={<Country />} />
      </Routes>
    </>  
);

export default App;
