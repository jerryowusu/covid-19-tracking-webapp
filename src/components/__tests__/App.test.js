import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { createStore, combineReducers, applyMiddleware } from 'redux';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import App from '../../App';
import covidReducer from '../../redux/covidData';

const reducer = combineReducers({
  covidReducer,
});

const store = createStore(
  reducer,
  applyMiddleware(logger, thunk),
);

describe('Render App', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <App />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
