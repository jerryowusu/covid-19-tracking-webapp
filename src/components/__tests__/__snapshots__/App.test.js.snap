// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App renders correctly 1`] = `
Array [
  <nav
    className="navbar"
  >
    <span
      className="nav-date"
    >
      2022
    </span>
    <span
      className="nav-title"
    >
      covid 19
      /
    </span>
  </nav>,
  <section>
    <header
      className="header"
    >
      <img
        alt="world-map"
        className="img"
        height="60%"
        src="worldmap.png"
        width="60%"
      />
      <div
        className="wrapper"
      >
        <h1
          className="header-title"
        >
          World
        </h1>
        <h3
          className="header-number"
        >
          444,638,071
        </h3>
      </div>
    </header>
    <form
      className="search-form"
    >
      <span
        className="form-span"
      >
        Infections by Country
      </span>
      <input
        className="search-input"
        name="search"
        onChange={[Function]}
        placeholder="country"
        type="text"
      />
    </form>
    <ul
      className="all-cards"
    />
  </section>,
]
`;
